### ISSUE TITLE
Unjustified(?) unbound var warning in try exception statements

### ISSUE THREAD

**Issue by abelcheung** [microsoft/pylance-release#913]
## Environment data

-   Language Server version: 2021.2.0 or newer
-   OS and version: XXX
-   Python version (& distribution if applicable, e.g. Anaconda): XXX

## Issue

Code sample below would cause Pylance to issue type check error, please see screenshot further down. I think this is a regression, since Pylance 2021.1.3 passes checking for this code snippet.

## Code Sample
```python
try:
    # any statement will do
    _ = 'text'.index('a')
except:
    var = 1
else:
    var = 2
finally:
    print(var)
```

## Screenshot

![圖片](https://user-images.githubusercontent.com/83110/106845304-937b5280-66a2-11eb-9785-8f10b0542d8e.png)

**Comment by erictraut** [comment]
I agree this is a false positive error. It arises from a fix for another class of false positives that previously occurred with try/except/else/finally statements. 

In this _very_ specific case, where there is both an "else" clause and a "bare except" clause (i.e. an except clause with no specified exception type) and a finally clause, we can avoid this particular false positive. I'll make the change to fix this specific case. I'll point out that if you add an exception type to the except clause (e.g. `except NameError:`), then this is a legitimate error because an exception of a different type could directly execute the `finally` clause.

**Comment by abelcheung** [comment]
Indeed you're correct about the other non-corner cases. Thanks for the prompt triaging!

**Comment by erictraut** [comment]
This will be fixed in the next release of pylance. 

Here's the PR in case you're interested: https://github.com/microsoft/pyright/commit/5bf5640335e7c76cabb07bace3d3d89d93d558f4. I used your sample in the unit test. :)

Thanks for the bug report.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.2.1, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202121-10-february-2021
